packages/ckeditor5-ui/tests/bindings/addKeyboardHandlingForGrid.js

```diff
@@ -0,0 +1,274 @@
+			//         [ ][ ][ ]	       [ ][ ][ ]
+			//         [x]      	       [ ]
+			const keyEvtData = {
+				keyCode: keyCodes.arrowdown,
+				preventDefault: sinon.spy(),
+				stopPropagation: sinon.spy()
+			};
+
+			focusTracker.focusedElement = gridElementsCollection.get( 6 ).element;
+
+			const spy = sinon.spy( gridElementsCollection.first, 'focus' );
+
+			keystrokes.press( keyEvtData );
+			sinon.assert.calledOnce( spy );
+		} );
+
+		it( 'arrowup at the first row moves focus to the one before last row if here is no item in the last row for this column', () => {
+			// before: [ ][x][ ]	after: [ ][ ][ ]	key: ↑
+			//         [ ][ ][ ]	       [ ][x][ ]
+			//         [ ]      	       [ ]
+			const keyEvtData = {
+				keyCode: keyCodes.arrowup,
+				preventDefault: sinon.spy(),
+				stopPropagation: sinon.spy()
+			};
+
+			focusTracker.focusedElement = gridElementsCollection.get( 1 ).element;
+
+			const spy = sinon.spy( gridElementsCollection.get( 4 ), 'focus' );
+
+			keystrokes.press( keyEvtData );
+			sinon.assert.calledOnce( spy );
+		} );
+
+		it( 'arrowdown at the one before last row moves focus to the first row if here is no item in the last row for this column', () => {
+			// before: [ ][ ][ ]	after: [ ][x][ ]	key: ↓
+			//         [ ][x][ ]	       [ ][ ][ ]
+			//         [ ]      	       [ ]
+			const keyEvtData = {
+				keyCode: keyCodes.arrowdown,
+				preventDefault: sinon.spy(),
+				stopPropagation: sinon.spy()
+			};
+
+			focusTracker.focusedElement = gridElementsCollection.get( 4 ).element;
+
+			const spy = sinon.spy( gridElementsCollection.get( 1 ), 'focus' );
+
+			keystrokes.press( keyEvtData );
+			sinon.assert.calledOnce( spy );
+		} );
+	} );
+
+	describe( 'first and last item', () => {
+		it( 'arrowleft moves focus to the last grid item if the first one was focused', () => {
+			// before: [x][ ][ ]	after: [ ][ ][ ]	key: ←
+			//         [ ][ ][ ]	       [ ][ ][ ]
+			//         [ ]      	       [x]
+			const keyEvtData = {
+				keyCode: keyCodes.arrowleft,
+				preventDefault: sinon.spy(),
+				stopPropagation: sinon.spy()
+			};
+
+			focusTracker.focusedElement = gridElementsCollection.first.element;
+
+			const spy = sinon.spy( gridElementsCollection.last, 'focus' );
+
+			keystrokes.press( keyEvtData );
+			sinon.assert.calledOnce( spy );
+		} );
+
+		it( 'arrowright moves focus to the first grid item if the last one was focused', () => {
+			// before: [ ][ ][ ]	after: [x][ ][ ]	key: →
+			//         [ ][ ][ ]	       [ ][ ][ ]
+			//         [x]      	       [ ]
+			const keyEvtData = {
+				keyCode: keyCodes.arrowright,
+				preventDefault: sinon.spy(),
+				stopPropagation: sinon.spy()
+			};
+
+			focusTracker.focusedElement = gridElementsCollection.last.element;
+
+			const spy = sinon.spy( gridElementsCollection.first, 'focus' );
+
+			keystrokes.press( keyEvtData );
+			sinon.assert.calledOnce( spy );
+		} );
+	} );
+} );
+
+class TestView extends View {
```

Comment:
Better keep it in the top-level `describe()`. Don't remember exactly how but these could leak somehow and affect other tests in some cases.